src/main/java/edu/umich/ctools/sectionsUtilityTool/FriendServlet.java

```diff
@@ -17,235 +17,353 @@
+            M_log.error(json.toString());
+        }
+        out.print(json.toString());
+        out.flush();
+
+        long stopTime = System.currentTimeMillis();
+        long elapsedTime = stopTime - startTime;
+        M_log.info(String.format("Social login account creation Api response took %sms", elapsedTime));
+
+    }
+
+    private StringEntity socialLoginApiRequestBody(String spURL, String nonUmichEmail, String friendFirstName, String friendLastName) throws UnsupportedEncodingException {
+        JSONObject apiPostBody = new JSONObject();
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SP_ENTITY_ID, spURL);
+        //Legacy required parameter in cirrusAPI will be deprecated in future so need to be part of the API but can be any value
+        apiPostBody.put(Utils.SOCIAL_LOGIN_CLIENT_REQUEST_ID, "request001");
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SERVICE_NAME, "UMICH Invite");
+        apiPostBody.put(Utils.SOCIAL_LOGIN_EMAIL_ADDRESS, nonUmichEmail);
+        apiPostBody.put(Utils.SOCIAL_LOGIN_EMAIL_SUBJECT, String.format("UMICH Invitation for %s %s" ,friendFirstName, friendLastName));
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SPONSOR_EPPN, Utils.FRIEND_CONTACT_EMAIL);
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SPONSOR_MAIL, Utils.FRIEND_CONTACT_EMAIL);
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SPONSOR_GIVENNAME, "ITS");
+        apiPostBody.put(Utils.SOCIAL_LOGIN_SPONSOR_SURNAME, "Service Center");
+        StringEntity stringEntity = new StringEntity(apiPostBody.toString());
+        return stringEntity;
+    }
+
+    /*
+     * This method is handling all the different Api request like PUT, POST etc to Friend Accounts.
+     */
+    private void friendRestApiCall(HttpServletRequest request,
+                                   HttpServletResponse response){
+        M_log.debug("friendRestApiCall(): called");
+        long startTime = System.currentTimeMillis();
+        JsonObject json;
+        PrintWriter out = null;
+        try {
+            request.setCharacterEncoding(Utils.CONSTANT_UTF_8);
+            Friend myFriend = new Friend();
+            response.setContentType(Utils.APPLICATION_JSON);
+            Properties appExtSecureProperties = Friend.appExtSecureProperties;
+
+            if (appExtSecureProperties == null) {
+                M_log.error("Failed to load system properties(sectionsToolProps.properties) for SectionsTool");
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                out = response.getWriter();
+                json = buildResponseObject("error", appExtProperties.getProperty("property.file.load.error"), "");
+                out.print(json.toString());
+                out.flush();
+                return;
+            }
+            Utils.logApiCall(request);
+            friendApiConnectionLogic(request, response, myFriend);
+        } catch (IOException e) {
+            response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
```

Comment:
The exception could be generalized into one single function, to handle both IOException and other Exception types

Reply:
OK, I have generalized the Exception to avoid code repetition